src/proper_gen.erl

```diff
@@ -418,6 +420,53 @@ atom_gen(Size) ->
 atom_rev(Atom) ->
     {'$used', atom_to_list(Atom), Atom}.
 
+%% @private
```

Comment:
These two functions, and the PR in general, should describe the main idea of the existing atom generation algorithm and the assumptions it relies on.  Also its limitations.

Having written that, if I understand what the `existing_atom_gen()` does, it is in fact **not** an atom generator (in the sense that `proper_gen:atom_gen/1` is), but it simply returns a random atom from those (currently) loaded in the atom table, which it stores in its own map, right?

Since there is a _magic_ way (the `<<131, 75, Index:24>>` trick) to access each of these atoms given an `Index`, and the upper limit of this index is known ( it is given by `erlang:system_info(atom_count)`), what is the point of first iterating through all these atoms putting them in an internal map, rather than randomly picking one `Index` in the range and returning the corresponding atom?  (Perhaps after a `num_tries` if this process is unlucky to only end up in atoms starting with `$`.)   What I am missing?

Incidentally, I do not like the magic constants 131, 75 and 24.  They should be made appropriate defines and ideally obtained by the Erlang/OTP version somehow.

Reply:
> Having written that, if I understand what the `existing_atom_gen()` does, it is in fact **not** an atom generator (in the sense that `proper_gen:atom_gen/1` is), but it simply returns a random atom from those (currently) loaded in the atom table, which it stores in its own map, right?

That is correct, yes. "Find all existing atoms, (store them in the map,) pick one at random". The term "generator" is probably not the exact term to use here (I guess this is you objection?), but what _is_ the correct term to use?

> What I am missing?

Nothing I think ;) That is a great suggestion, I'll do that. How about we use a fallback atom like `''` to return if all tries turn up atoms starting with `$`? We need to return something, right?